internal/resources/metal/vlan/datasource_test.go

```diff
@@ -335,3 +291,53 @@ func testAccMetalDatasourceVlanCheckDestroyed(s *terraform.State) error {
 
 	return nil
 }
+
+func TestAccDataSourceMetalVlan_byVxlanMetro_upgradeFromVersion(t *testing.T) {
+	rs := acctest.RandString(10)
+	metro := "sv"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheckMetal(t) },
```

Comment:
This test is failing because of differences between the SDKv2 resource state and the Framework resource state:

```
=== NAME  TestAccDataSourceMetalVlan_byVxlanMetro_upgradeFromVersion
    datasource_test.go:299: Step 2/2 error: Pre-apply plan check(s) failed:
        expected empty plan, but data.equinix_metal_vlan.bardsvlan has planned action(s): [read]
        expected empty plan, but data.equinix_metal_vlan.dsvlan has planned action(s): [read]
        expected empty plan, but equinix_metal_vlan.barvlan has planned action(s): [delete create]
        expected empty plan, but equinix_metal_vlan.foovlan has planned action(s): [delete create]
```

We'll need to look through the old schema and the new schema and try to figure out which attributes could be different.  This might be a matter of missing/different default values; missing `UseStateForUnknown`; Framework resource storing an attribute as `""` whereas SDKv2 resource stored it as `nil` (or vice versa).

I haven't found a way to get the acceptance test to print the plan to stdout, but maybe there is a way to do that?  Another way to dig into problems like this is to temporarily print the state and plan to stdout in the resource's `Update` function and run this specific test to see what the differences are.

Reply:
handled this by adding a default value in facility